### PR TITLE
[ci] Use transaction strategy for database_cleaner

### PR DESCRIPTION
**Pull request by dmarcoux**
This is an attempt to use a faster strategy for database_cleaner. Refer to the [database_cleaner's README](https://github.com/DatabaseCleaner/database_cleaner#what-strategy-is-fastest) for details.

Locally, I had a big speed improvement when running a single spec (**transaction - 38.7s, truncation - 1min 50s**). However, I don't know if the whole test suite will still be green with this change.

### CODE REVIEW

**Comment by DavidKang** [comment]
@dmarcoux , let's see :smile_cat:

**Comment by coolo** [comment]
Note that database_cleaner was introduced all together to have truncation for feature tests:

https://github.com/dmarcoux/open-build-service/commit/be29298efa187dc14f5956a3f31de7ae1333de34

If you use transaction strategy (and pass test suite), you don't need database cleaner and can just revert to what was there before henne removed it. See:
https://anti-pattern.com/transactional-fixtures-in-rails

I assume feature tests are fine with using transactions since quite a while - otherwise all the mocking and stubing done by rspec-mock wouldn't work.

**Comment by coolo** [comment]
ok, it's not so easy - and it isn't quicker without database_cleaner :)

File: src/api/spec/support/database_cleaner.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@
     # Before the test suites runs make sure the database is empty
     log_level = Rails.logger.level
     Rails.logger.level = :fatal
-    DatabaseCleaner.clean_with(:truncation)
+    DatabaseCleaner.clean_with(:transaction)
```
Comment (coolo) [inline src/api/spec/support/database_cleaner.rb]: this doesn't make sense, sorry. The code tries to cleanup the initial database - and you simply can't clean up existing data with a transaction.

The code here is a little conservative and for your use case it's most likely preferable to just remove it completly by means of an environment variable.

**Comment by ChrisBr** [comment]
cc @hennevogel IIRC there was a reason why we used ``truncation`` instead of ``transaction``, right?

**Comment by hennevogel** [comment]
@ChrisBr jupp, that are are not on the same thread as the database.

**Comment by ChrisBr** [comment]
> jupp, that are are not on the same thread as the database. See the comments in the initial commit (be29298)

Yeah remembering that but wondering why tests are now passing then?

**Comment by coolo** [comment]
because the code @dmarcoux changed is only for the initial cleanup on suite start - the CI has nothing in there, so the CI won't that he changed it to a NOP.

**Pull request closed without merging**